cmd/thanos/bucket.go

```diff
@@ -472,6 +472,8 @@ func download(ctx context.Context, logger log.Logger, bkt objstore.Bucket, fetch
 		return nil, err
 	}
 
+	blocks = []metadata.Meta{}
```

Comment:
Also, if we are doing this then converting this and `err` into "ordinary" variables seems like a nice refactoring that you could do since you'll revise this PR either way.

```suggestion
	blocks := []metadata.Meta{}
```

And then remove it from the function's definition. :P Would look cleaner, right?

Reply:
Yes, completely. I will do this.